### PR TITLE
add task solution

### PR DESCRIPTION
**Pull request by Daniil-A4**
- [DEMO LINK](https://Daniil-A4.github.io/layout_product-cards/)
- [TEST REPORT LINK](https://Daniil-A4.github.io/layout_product-cards/report/html_report/)

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -5,9 +5,47 @@
...
+    >
+    <link rel="stylesheet" href="./styles/index.scss">
   </head>
   <body>
-    <h1>Product cards</h1>
+    <div class="card" data-qa="card">
+      <img class="card__img"
+        src="images/imac.jpeg"
+        alt="imac"
+      >
```
Comment (Viktor-Kost) [inline src/index.html:22]: apply code styling recommendations from a checklist. The same issue is for `<a>` tag
```suggestion
      <img 
        class="card__img"
        src="images/imac.jpeg"
        alt="imac"
      >
```

File: src/index.html [hunk 2]
```diff
@@ -5,9 +5,47 @@
...
+
+      <div class="card__box">
+        <ul class="stars stars--2">
+          <li class="stars__star"></li>
+          <li class="stars__star"></li>
+          <li class="stars__star"></li>
+          <li class="stars__star"></li>
+          <li class="stars__star"></li>
+        </ul>
+        <p class="card__revie">Reviews: 5</p>
```
Comment (Viktor-Kost) [inline src/index.html:35]:
```suggestion
        <p class="card__review">Reviews: 5</p>
```

File: src/styles/blocks/card.scss [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    line-height: 112.5%;
+  }
+
+  &__btn {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    text-decoration: none;
+    width: 100%;
+    height: 40px;
```
Comment (Viktor-Kost) [inline src/styles/blocks/card.scss:68]: it's better not hardcode height for elements with content inside. Just use padding: 11px 0;

File: src/styles/blocks/card.scss [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    border-radius: 5px;
+    border: 1px solid $blueColor;
+    background: $blueColor;
+    font-size: 14px;
+    font-weight: 700;
+    text-transform: uppercase;
+    color: #fff;
+    cursor: pointer;
+    box-sizing: border-box;
+    line-height: 114.286%;
```
Comment (Viktor-Kost) [inline src/styles/blocks/card.scss:78]:
```suggestion
    line-height: 16px;
```

File: src/styles/blocks/stars.scss [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    background-repeat: no-repeat;
+    width: 16px;
+    height: 16px;
+
+    &:not(:last-child) {
+      margin-right: 4px;
+    }
+  }
+
+  &--2 .stars__star:nth-child(-n + 4) {
```
Comment (Viktor-Kost) [inline src/styles/blocks/stars.scss:19]: add styles for all possible cases to make it flexible (just copy styles from stars task

File: src/styles/blocks/card.scss [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+    font-size: 12px;
+    font-weight: 400;
+    color: $greyColor;
+    line-height: 150%;
+  }
+
+  &__price {
+    font-weight: 700;
+    color: $mainColor;
+    line-height: 112.5%;
```
Comment (Viktor-Kost) [inline src/styles/blocks/card.scss:59]: use integers

File: src/styles/blocks/variables.scss [hunk 7]
```diff
@@ -0,0 +1,3 @@
+$mainColor: #060b35;
+$greyColor: #616070;
+$blueColor: #00acdc;
```
Comment (Viktor-Kost) [inline src/styles/blocks/variables.scss:3]: it'd be more flexible
```suggestion
$neutralColor: #616070;
$accentColor: #00acdc;
```

**Review by Viktor-Kost — CHANGES_REQUESTED** [review]

File: src/styles/blocks/card.scss (hunk 3 above)
Comment (Daniil-A4) [reply src/styles/blocks/card.scss:68]: Which difference between these two options: 'height: 40px; and padding: 11px 0;'? I think using second option is hardcode too?

File: src/styles/index.scss [hunk 8]
```diff
@@ -1,3 +1,17 @@
-body {
+html {
+  font-family: "Roboto", sans-serif;
```
Comment (denys-danyliuk) [inline src/styles/index.scss:2]: 2'nd point in checklist

File: src/styles/blocks/stars.scss [hunk 9]
```diff
@@ -0,0 +1,26 @@
+.stars {
+  list-style: none;
+  padding: 0;
+  margin: 0;
```
Comment (denys-danyliuk) [inline src/styles/blocks/stars.scss:4]: don't add external styles to bem block

File: src/index.html [hunk 10]
```diff
@@ -5,9 +5,49 @@
...
+    >
+    <link rel="stylesheet" href="./styles/index.scss">
   </head>
   <body>
-    <h1>Product cards</h1>
+    <div class="card" data-qa="card">
+      <img
+        class="card__img"
+        src="images/imac.jpeg"
+        alt="imac"
```
Comment (denys-danyliuk) [inline src/index.html:22]: 13'th point in checklist

File: src/index.html [hunk 11]
```diff
@@ -5,9 +5,49 @@
     <meta name="viewport"
           content="width=device-width, user-scalable=no, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0">
     <title>Product cards</title>
-    <link rel="stylesheet" href="./style.css">
+    <link rel="preconnect" href="https://fonts.googleapis.com">
+    <link rel="preconnect"
+      href="https://fonts.gstatic.com"
+      crossorigin="anonymous"
+    >
```
Comment (denys-danyliuk) [inline src/index.html:12]: 13'th point

**Review by denys-danyliuk — CHANGES_REQUESTED** [review]

**Review by denys-danyliuk — APPROVED** [review]

**Pull request closed without merging**